### PR TITLE
libwebp QNX fix

### PR DESCRIPTION
**Pull request by blackliner**
Specify library name and version:  **libwebp**

Just curious, is it ok to have QNX compatibility in CCI?

upstream fix: https://bugs.chromium.org/p/webp/issues/detail?id=498

- 👍 I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- 👍 I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- 👍 I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- 👍 I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
https://docs.conan.io/en/1.31/integrations/cross_platform/qnx_neutrino.html

Yes?

**Review by prince-chrismc — DISMISSED** [review]

**Comment by blackliner** [comment]
> https://docs.conan.io/en/1.31/integrations/cross_platform/qnx_neutrino.html
> 
> Yes?

yes. I previously had a check in it for `self.settings.os == "Neutrino"`, but a CCI hook showed an error about "Neutrino might not be supported" or so. This way CMake does the check

**Comment by ericLemanissier** [comment]
IIUC, this change comes from https://chromium-review.googlesource.com/c/webm/libwebp/+/2637274
Does it work if you download the patch and apply it using with `tools.patch` ? It makes maintenance easier for future versions (but it's not blocking for this PR)

**Comment by blackliner** [comment]
No, the patch does unfortunately not work, because their file on master has already some changes to 1.1.0

**Comment by prince-chrismc** [comment]
Should it be applied to the older version as well?

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
#3903 please add a link to the upstream (the one eric shared works)

**Comment by blackliner** [comment]
> #3903 please add a link to the upstream (the one eric shared works)

I posted a link in the OP, is that what you want?

**Comment by prince-chrismc** [comment]
Sorry, that was not clear, it the patch itself pretty please 🙏 this was when we go to bump and we see it in the patch we know it's no longer required

**Comment by blackliner** [comment]
there is a comment in the patchfile also: `# work around cmake bug on QNX (https://cmake.org/Bug/view.php?id=11333)`

**Review by prince-chrismc — DISMISSED** [review]
LGTM!

File: recipes/libwebp/all/patches/0002-qnx.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
+diff --git a/cmake/deps.cmake b/cmake/deps.cmake
+index f19c0378..7b2d7cd7 100644
+--- a/cmake/deps.cmake
++++ b/cmake/deps.cmake
+@@ -24,7 +24,8 @@ check_c_source_compiles("
+ # Check for libraries.
+ find_package(Threads)
+ if(Threads_FOUND)
+-  if(CMAKE_USE_PTHREADS_INIT)
++  # work around cmake bug on QNX (https://cmake.org/Bug/view.php?id=11333)
```
Comment (ericLemanissier) [inline recipes/libwebp/all/patches/0002-qnx.patch:10]:
```suggestion
+  # work around cmake bug on QNX (https://chromium-review.googlesource.com/c/webm/libwebp/+/2637274)
```
this cmake issue is useful to understand the bug, unfortunately it does not give the information of which version of libwbp already have this fix. deployment of future versions will be easier if the patch links to the upstream issue directly.

Comment (blackliner) [reply recipes/libwebp/all/patches/0002-qnx.patch:10]: Thats all I got, the CMake issue also fixes the QNX, read the gerrit thread if you are interested

**Review by ericLemanissier — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**